dracut/31decrypt-root/decrypt-root

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+# Check if there is a partition labeled ROOT and, if so, determine its UUID.
+match=$(lsblk -o UUID,PARTLABEL | grep -w ROOT)
+if [[ $? -ne 0 ]]; then
+    exit 0
+fi
+uuid="${match%% *}"
+
+# Check if the partition labeled ROOT is a LUKS device.
+IFS= read -r -n "4" header < /dev/disk/by-uuid/$uuid
+if [[ "$header" != "LUKS" ]]; then
+    exit 0
+fi
+
+# Generate a systemd-cryptsetup unit to decrypt the root partition.
+# We call the LUKS device `rootencrypted`, no matter what it was called in the Ignition configuration.
+echo "rootencrypted UUID=$uuid none luks" > /tmp/crypttab
```

Comment:
```suggestion
echo "rootencrypted UUID=$uuid none luks,tpm2-device=auto" > /tmp/crypttab
```
Would be nice to do this for systemd-cryptsetup to also try it's luck based on the LUKS metadata. I didn't manage yet to get it work but it should be possible.

Reply:
I added it. (It should not be needed because `clevis` is ready to respond to systemd-cryptsetup's prompts, but it also shouldn't hurt.)

Reply:
There were failing Tang and TPM tests on arm64 in the test run you triggered, but that's expected because it ran against the mantle main branch. Only the modified tests [here](https://github.com/flatcar/mantle/pull/511) are expected to pass because the `rd.luks.*` cmdline parameters in the current tests create a failing unit in the initramfs.

I downloaded the arm64 image from the pipeline and ran the updated Tang and TPM tests locally. The tests all passed. Could we maybe retrigger the pipeline to run against the tests from the mantle PR?

```
mantle on remove-luks-cmdline ❯ sudo ./bin/kola run --board arm64-usr -v -k -b cl -p qemu --qemu-image ./flatcar_production_qemu_uefi_image.img --qemu-bios=./flatcar_production_qemu_uefi_efi_code.fd cl.tang.* cl.tpm.*
2024-03-25T19:28:50Z cli: Started logging at level INFO
2024-03-25T19:28:50Z util: dnsmasq[24664]: started, version 2.89 cachesize 150
2024-03-25T19:28:50Z util: dnsmasq[24664]: compile time options: IPv6 GNU-getopt DBus no-UBus no-i18n IDN DHCP DHCPv6 no-Lua TFTP conntrack ipset nftset auth cryptohash DNSSEC loop-detect inotify dumpfile
2024-03-25T19:28:50Z util: dnsmasq[24664]: warning: no upstream servers configured
2024-03-25T19:28:50Z util: dnsmasq-dhcp[24664]: DHCP, static leases only on 10.0.0.1, lease time 1h
2024-03-25T19:28:50Z util: dnsmasq-dhcp[24664]: DHCPv4-derived IPv6 names on fd00::
2024-03-25T19:28:50Z util: dnsmasq-dhcp[24664]: router advertisement on fd00::
2024-03-25T19:28:50Z util: dnsmasq-dhcp[24664]: IPv6 router advertisement enabled
2024-03-25T19:28:50Z util: dnsmasq[24664]: cleared cache
2024-03-25T19:28:50Z network/ntp: Started NTP server on [::]:123
2024-03-25T19:28:50Z kola: Creating cluster to check semver...
2024-03-25T19:30:43Z network/ntp: Recieved NTP request from 10.0.0.2:44469
2024-03-25T19:31:15Z network/ntp: Recieved NTP request from 10.0.0.2:60090
2024-03-25T19:31:20Z kola: Using "3916.0.0+nightly-20240321-2100-1-g448d7c8cb3" as version to filter tests...
=== RUN   cl.tpm.nonroot
=== RUN   cl.tang.root
=== RUN   cl.tpm.root
=== RUN   cl.tang.nonroot
2024-03-25T19:34:04Z network/ntp: Recieved NTP request from 10.0.0.3:36340
2024-03-25T19:34:32Z kola/cluster: SSH: running command: lsblk -o NAME,LABEL
2024-03-25T19:34:33Z kola/cluster: SSH: running command: lsblk --json
2024-03-25T19:34:36Z network/ntp: Recieved NTP request from 10.0.0.3:58827
2024-03-25T19:36:34Z network/ntp: Recieved NTP request from 10.0.0.3:39558
2024-03-25T19:36:57Z kola/cluster: SSH: running command: lsblk -o NAME,LABEL
2024-03-25T19:36:59Z kola/cluster: SSH: running command: lsblk --json
--- PASS: cl.tpm.nonroot (342.27s)
2024-03-25T19:39:44Z network/ntp: Recieved NTP request from 10.0.0.4:53216
2024-03-25T19:40:12Z kola/cluster: SSH: running command: lsblk -o NAME,LABEL
2024-03-25T19:40:13Z kola/cluster: SSH: running command: lsblk --json
2024-03-25T19:40:16Z network/ntp: Recieved NTP request from 10.0.0.4:35474
2024-03-25T19:42:20Z network/ntp: Recieved NTP request from 10.0.0.4:38001
2024-03-25T19:42:43Z kola/cluster: SSH: running command: lsblk -o NAME,LABEL
2024-03-25T19:42:45Z kola/cluster: SSH: running command: lsblk --json
--- PASS: cl.tang.nonroot (346.22s)
        tang.go:149: Started tang on 10.0.0.1:45109
2024-03-25T19:45:26Z network/ntp: Recieved NTP request from 10.0.0.5:58449
2024-03-25T19:45:58Z network/ntp: Recieved NTP request from 10.0.0.5:57068
2024-03-25T19:45:59Z kola/cluster: SSH: running command: lsblk -o NAME,LABEL
2024-03-25T19:46:01Z kola/cluster: SSH: running command: lsblk --json
2024-03-25T19:48:21Z kola/cluster: SSH: running command: lsblk -o NAME,LABEL
2024-03-25T19:48:22Z kola/cluster: SSH: running command: lsblk --json
--- PASS: cl.tpm.root (337.51s)
2024-03-25T19:53:43Z network/ntp: Recieved NTP request from 10.0.0.6:56679
2024-03-25T19:54:11Z kola/cluster: SSH: running command: lsblk -o NAME,LABEL
2024-03-25T19:54:13Z kola/cluster: SSH: running command: lsblk --json
2024-03-25T19:54:15Z network/ntp: Recieved NTP request from 10.0.0.6:56049
2024-03-25T19:56:21Z network/ntp: Recieved NTP request from 10.0.0.6:38254
2024-03-25T19:56:51Z kola/cluster: SSH: running command: lsblk -o NAME,LABEL
2024-03-25T19:56:53Z kola/cluster: SSH: running command: lsblk --json
2024-03-25T19:56:53Z network/ntp: Recieved NTP request from 10.0.0.6:41550
--- PASS: cl.tang.root (510.97s)
        tang.go:149: Started tang on 10.0.0.1:34615
PASS, output in _kola_temp/qemu-2024-03-25-2028-24642
```

Reply:
I see you ran the arm64 tests, so I guess it's good then without doing one more run here.